File: lingting-spring-core/src/main/java/live/lingting/spring/post/ContextComposeBeanPostProcessor.java
package live.lingting.spring.post;

import live.lingting.framework.context.ContextComponent;
import org.slf4j.Logger;

/**
 * @author lingting 2022/10/22 15:10
 */
public class ContextComposeBeanPostProcessor implements SpringBeanPostProcessor {

	private static final Logger log = org.slf4j.LoggerFactory.getLogger(ContextComposeBeanPostProcessor.class);

	@Override
	public boolean isProcess(Object bean, String beanName, boolean isBefore) {
		return bean instanceof ContextComponent;
	}

	@Override
	public Object postProcessAfter(Object bean, String beanName) {
		log.trace("class [{}] start.", bean.getClass());
		((ContextComponent) bean).onApplicationStart();
		return bean;
	}

}

File: lingting-spring-core/src/main/kotlin/live/lingting/spring/post/ContextComposeBeanPostProcessor.kt
package live.lingting.spring.post

import live.lingting.framework.context.ContextComponent
import live.lingting.framework.util.Slf4jUtils.logger

/**
 * @author lingting 2022/10/22 15:10
 */
class ContextComposeBeanPostProcessor : SpringBeanPostProcessor {
    override fun isProcess(bean: Any, beanName: String, isBefore: Boolean): Boolean {
        return bean is ContextComponent
    }

    override fun postProcessAfter(bean: Any, beanName: String): Any {
        log.trace("class [{}] start.", bean.javaClass)
        (bean as ContextComponent).onApplicationStart()
        return bean
    }

    companion object {
        private val log = logger()
    }
}
